### PR TITLE
armv7a architecture specifics in arch/armv7a/ps_archxxx.h

### PR DESCRIPTION
**Pull request by phanikishoreg**
Merged ps_arch.h from Runyu's stuff into the latest, fixed some for style and added configuration option for build configuration.

### CODE REVIEW

File: plat/arch/cav7/ps_arch_cav7_common.h [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+ps_cas(unsigned long *target, unsigned long old, unsigned long updated)
+{
+	unsigned long oldval, res;
+
+	do {
+		oldval = ps_ldrexw(target);
+
+		if(oldval == old) {
+			/* 0=succeeded, 1=failed */
+			res=ps_strexw(updated, target);
```
Comment (gparmer) [inline plat/arch/cav7/ps_arch_cav7_common.h:94]: spaces around `=`

File: plat/arch/cav7/ps_arch_cav7_common.h [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+ */
+static inline int 
+ps_faa(int *var, int value)
+{
+	unsigned int res;
+	int oldval;
+
+	do {
+		oldval = (int) ps_ldrexw((volatile unsigned long *) var);
+		res    = ps_strexw((unsigned long) (oldval+value), (volatile unsigned long *) var);
```
Comment (gparmer) [inline plat/arch/cav7/ps_arch_cav7_common.h:116]: Spaces around +

File: plat/arch/cav7/ps_arch_cav7_common.h [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+#define ps_load(addr) (*(volatile __typeof__(*addr) *)(addr))
+#define ps_store(addr, val) ((*(volatile __typeof__(*addr) *)(addr)) = val)
+
+/*
+ * Only atomic on a uni-processor, so not for cross-core coordination.
+ * Faster on a multiprocessor when used to synchronize between threads
+ * on a single core by avoiding locking.
+ */
+static inline int
+ps_upcas(unsigned long *target, unsigned long old, unsigned long updated)
```
Comment (gparmer) [inline plat/arch/cav7/ps_arch_cav7_common.h:135]: This seems the same as ps_cas? If so, shouldn't the body just be `{ return ps_cas(target, old, updated); }`

File: plat/arch/cav7/ps_arch_cav7_common.h [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+		}
+	} while(res);
+
+	return 1;
+}
+
+static inline long
+ps_upfaa(unsigned long *var, long value)
+{
+	unsigned int res;
```
Comment (gparmer) [inline plat/arch/cav7/ps_arch_cav7_common.h:158]: Same comment.

**Comment by phanikishoreg** [comment]
Thanks for the feedback. I resolved them all.

**Comment by phanikishoreg** [comment]
@gparmer Addressed your feedback and moved arm stuff to armv7a now. Ready for merge whenever its convenient for you.

**Comment by gparmer** [comment]
Let me know when to merge. I assumed that you tested that x86 still works? As you made no general modifications, I'd assume so...

**Comment by phanikishoreg** [comment]
It is ready to merge. 

I've tested the compilation of the same for x86 through composite, and it works fine. As you mentioned, I touch nothing for x86.
However, I am still blocked on debugging the bootup issue of the Cos system on x86 after merging Runyu's platform stuff.

**Pull request merged**